modxcms/revolution
# [Port to 3.x] Fix Various Issues with Form Panel Validation and Field Display - Part 4, Template Panel

_NOTE: This is part 4 of 6 of a revised submission of #15887. This should be reviewed after part 2 is merged (there are two small changes needed from that PR to make the locked switch appear correctly)._

### What does it do?

Ports enhancements to the Template form panel from PR #15146 to 3.x (including subsequent fixes by others). 
Changes include:
1. Updated panel JS to make form layout consistent with the layout introduced in the current TV panel, and utilize new shared methods introduced in the main FormPanel via part 1 of this PR set (already merged).
2. Added a secondary Media Source combo for the preview image, as the new layout makes the main Media Source combo available _only_ when Is Static is ticked. For the purposes of this PR, the new combo mirrors the original (so when one changes, the other reflects that same change) — this is, in effect, the current behavior, but it may be worthwhile to consider having a fully-independent secondary Source setting in the future.
3. A handful of minor lexicon updates.

### Why is it needed?

Makes needed UI and consistency improvements to the Template panel.

### How to test

1. From your terminal app, run `grunt build` from within the `_build/templates/default` folder and clear your browser cache.
2. Create and update a few Templates to verify behavior and appearance is as expected.

### Related issue(s)/PR(s)
Partial port of #15146
